core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -5706,6 +5706,13 @@ object ZIO extends ZIOCompanionPlatformSpecific {
       // TODO: Dotty doesn't infer this properly: io.bracket[R1, E1](a => UIO(a.close()))(use)
       acquireReleaseWith(io)(a => ZIO.succeed(a.close()))(use)
 
+    /**
+     * Like `withFinalizer, add a finalizer from AutoClosable.
+     */
+    def withFinalizerAutoCloseable[R1 <: R & Scope](implicit
```

Comment:
```scala
def withFinalizerAutoCloseable(implicit trace: ZTraceElement): ZIO[R with Scope, E, A]
```

Reply:
I think you could call this `withFinalizerAuto` for consistency with the old `toManagedAuto` and to be a little shorter.